aws/eks-anywhere
# Backport Controller Support for Reconciling Taints (#1175)

* vsphere provider checks if new kubeadmconfig template is needed based on WN config

* method to fetch kubeadmconfigtemplate

* regenerate kubeadmconfigtemplate spec if neccessary

* controller tests, testdata, and mocks

* update vsphere md template to accomodate mutable kubeadmconfigtemplate name

* vsphere provider tests, testdata, and mocks

* docker, tink and generic provider methods to support kubeadmconfigtemplate changes

* gofmt controller tests

* add kubeadmconfigtemplate to scheme

* check if new kubeadmconfigtemplate is needed during upgrade

* standardize method calls in templater

* update tests, mocks

* shorten kubeadmconfigtemplate generated name

* make sure to pass kubeadmconfigtemplate during upgrade

* update vsphere multiworkernode tests to properly pull the machine deployment associated with the wngc

* remove redundant defaulting behavior in vsphere template builder

* fix mock method signatures after merge

*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
